docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+in step 1.
+![AccountListObjectDiagram1](images/AccountListObjectDiagram3.png)
+
+Step 4: The user passes the command `delete-account CURRENCY`, for example `delete-account SGD`.
+
+![AccountListObjectDiagram4](images/AccountListObjectDiagram4.png)
+
+The following sequence diagram shows how the Create Account operation works
+![AccountListObjectDiagram1](images/CreateAccountSeqDiagram.png)
+
+### Delete-account feature
+
+### Add/Withdraw money feature
+
+The add money(deposit) and withdraw money feature is facilitated by `AddCommand` and `WithdrawCommand` which both
+extends the `Command` class. With the provided input from user (`CURRENCY` and `AMOUNT`), `AddCommand`
+and `WithdrawCommand`
+update the balance of respective currency account accordingly.
+
+Step 1. The newly created `SGD` account has an initial balance of 0
+![AddWithdrawCommandObjectDiagram1](images/AddWithdrawCommandObjectDiagram1.png)
+
+Step 2. The user passes command `add CURRENCY AMOUNT` (eg. `add SGD 100`), where `CURRENCY` must be one of the available
+currency and `AMOUNT` must be positive numbers.
+![AddWithdrawCommandObjectDiagram2](images/AddWithdrawCommandObjectDiagram2.png)
+
+Step 3. The user passes command `withdraw` (eg. `withdraw SGD 25`), where `AMOUNT` must be smaller than the currency
+account balance.
+![AddWithdrawCommandObjectDiagram3](images/AddWithdrawCommandObjectDiagram3.png)
+
+The following sequence diagram shows how the add money operation works.
+
+![](images/AddCommandSeqDiagram.png)
+
+The following sequence diagram shows how the money withdrawal operation works.
+
+![](images/WithdrawCommandSeqDiagram.png)
+
+### View balance feature
+
+The view balance feature is facilitated using `Account` instances stored within the `AccountList` object. The main
+functionality is to view the balance of a specific currency if the currency is specified, else view all the currencies
+in the account.
+
+![BalanceSequenceDiagram](images/BalanceSeqDiagram.png)
+
+### Show-rate feature
+
+The exchange feature is facilitated using two `Forex` instances to represent the
+exchange rates between two currencies. The current implementation reads manual exchange
+rates from an online source. Future implementation will use an API to maintain up-to-date
+exchange rates.
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
+
+The show-rate command executes as follows
+
+- The initial and target currency are parsed from the input
+- If the user included an amount, the amount is also parsed
+- A `Forex` object showing the rate from the initial to the target currency is made
+- A `Forex` object showing the rate from the target to the initial currency is made
+- The Ui prints the exchanged amount if an amount was provided by the user for both rates
+- The Ui will print the unit rate both ways if no amount was provided
+
+The following sequence diagram shows how the Show Rate command works
+![ShowRateSeqDiagram](images/ShowRateSeqDiagram.png)
```

Comment:
Double check this, but it seems like the conditions for the `alt` blocks are missing in this diagram.